src/mainloop.rkt

```diff
@@ -138,21 +138,31 @@
       #;(exp ,exp-x ,log-x)
       #;(log ,log-x ,exp-x))))
 
-(define (taylor-alt altn loc)
+(define (taylor-alt altn loc repr)
   (define expr (location-get loc (alt-program altn)))
   (define vars (free-variables expr))
+
+  ;; resugar/desugaring
+  (define prec (representation-name repr))
+  (define var-precs (for/list ([var vars]) (cons var prec)))
+
   (if (or (null? vars) ;; `approximate` cannot be called with a null vars list
-          (not (equal? (type-of expr (*var-reprs*)) 'real)))
+          (not (set-member? '(binary64 binary32) (type-of expr repr (*var-reprs*)))))
       (list altn)
       (for/list ([transform-type transforms-to-try])
         (match-define (list name f finv) transform-type)
         (define transformer (map (const (cons f finv)) vars))
         (alt
-         (location-do loc (alt-program altn) (λ (expr) (approximate expr vars #:transform transformer)))
+         (location-do loc 
+                      (alt-program altn) 
+                      (λ (x) ; taylor uses older format, resugaring and desugaring needed
+                        (parameterize-expr
+                          (approximate (unparameterize-expr x) vars #:transform transformer)
+                          prec)))
          `(taylor ,name ,loc)
          (list altn)))))
 
-(define (gen-series!)
+(define (gen-series! repr)
```

Comment:
I think the representation is stored in `(*output-repr*)`. We should use that global variable. Right, what?! A global?! Well, `mainloop.rkt` is useful from the REPL for debugging, and globals make that a lot easier.